utilities/titandb/blob_file_builder.cc

```diff
@@ -8,16 +12,26 @@ void BlobFileBuilder::Add(const BlobRecord& record, BlobHandle* handle) {
 
   buffer_.clear();
   record.EncodeTo(&buffer_);
+
+  CompressionType compression;
+  auto output = Compress(compression_ctx_, buffer_,
+                         &compressed_buffer_, &compression);
   handle->offset = file_->GetFileSize();
-  handle->size = buffer_.size();
-  status_ = file_->Append(buffer_);
+  handle->size = output.size();
+
+  status_ = file_->Append(output);
+  if (ok()) {
+    char tailer[kBlobTailerSize];
+    tailer[0] = compression;
+    EncodeFixed32(tailer+1, crc32c::Value(output.data(), output.size()));
+    status_ = file_->Append(Slice(tailer, sizeof(tailer)));
```

Comment:
Maybe you can merge two append call into one

Reply:
But I need to allocate another buffer for that. Actually, the `WritableFileWriter` contains a buffer inside by default, which means that the tailer will be copied to the buffer inside anyway.